### ISSUE TITLE
'cstddef' not found

### ISSUE THREAD

**Issue by dsprenkels** [sgielen/cosix#2]
I get an error while compiling on:

```
vagrant@vagrant:~/cosix/build$ uname -a
Linux vagrant 4.7.0-cloudabi+ #1 SMP Fri Jul 21 12:03:48 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

The compiler moans about it not being able to find `cstddef`. I think I do indeed not have any `stddef.h` for i686. What did I miss in the installation of the build dependencies?

```
HEAD is now at 97121f8... Fix uninitialized memory and (blocking) behavior of read_keystrokes in terminal_impl.
vagrant@vagrant:~/cosix/build$ cmake -DCMAKE_BUILD_TYPE=Debug  -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchain-i686-elf.cmake ..
** Cross compiling for i686-unknown-cloudabi -- tests disabled, bare metal build enabled
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vagrant/cosix/build
vagrant@vagrant:~/cosix/build$ make
Scanning dependencies of target memory
[  0%] Building CXX object memory/CMakeFiles/memory.dir/allocation.cpp.obj
[  1%] Building CXX object memory/CMakeFiles/memory.dir/allocator.cpp.obj
[  1%] Building CXX object memory/CMakeFiles/memory.dir/page_allocator.cpp.obj
In file included from /home/vagrant/cosix/memory/page_allocator.cpp:3:
In file included from /home/vagrant/cosix/fd/process_fd.hpp:3:
In file included from /home/vagrant/cosix/fd/fd.hpp:6:
/home/vagrant/cosix/memory/smart_ptr.hpp:8:10: fatal error: 'cstddef' file not found
#include <cstddef>
         ^~~~~~~~~
1 error generated.
memory/CMakeFiles/memory.dir/build.make:110: recipe for target 'memory/CMakeFiles/memory.dir/page_allocator.cpp.obj' failed
make[2]: *** [memory/CMakeFiles/memory.dir/page_allocator.cpp.obj] Error 1
CMakeFiles/Makefile2:1734: recipe for target 'memory/CMakeFiles/memory.dir/all' failed
make[1]: *** [memory/CMakeFiles/memory.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by EdSchouten** [comment]
Have you installed `i686-unknown-cloudabi-cxx-runtime`?

**Comment by dsprenkels** [comment]
That's it! Thank you so much.

**Comment by sgielen** [comment]
I'll add this to the `README.md` so this won't happen to hypothetical future users :)

**Comment by sgielen** [comment]
Fixed by a93fe9560977bcc1cfccea2c708af4144a99128e. I accidentally used the wrong issue number in the commit :-(
